pkg/serve.go

```diff
@@ -2,59 +2,99 @@ package pkg
 
 import (
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"net/http"
+	//"reflect"
+	"strings"
+	"strconv"
 
+	"github.com/dustin/go-humanize"
 	"github.com/flanksource/commons/logger"
 	"github.com/flanksource/commons/net"
 	"github.com/flanksource/karina-ui/pkg/api"
-	"gopkg.in/yaml.v2"
+  	"gopkg.in/yaml.v2"
+  	v1 "k8s.io/api/core/v1"
+  	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 )
 
 var config = make(map[string]api.ClusterConfiguration)
+var clientset *kubernetes.Clientset
+var properties []api.Property
+var freshproperties []api.Property
+
 
 func Serve(resp http.ResponseWriter, req *http.Request) {
+
 	logger.Infof("🚀 Fetching data")
 	var clusters []api.Cluster
+	var canary api.Canarydata
+
 	for name, cluster := range config {
 
-		var canary api.Canarydata
-		canaryResp, err := net.GET(cluster.CanaryChecker)
-		if err != nil {
-			logger.Errorf("❌ Canary Check failed with %s", err)
+		//properties = []api.Property{}
+
+		fmt.Printf("start: %v\n\n", name)
+		
+		clientset, clientsetErr := GetClient(cluster.Kubeconfig)
+		if clientsetErr != nil {
+			logger.Errorf("❗ Get K8s client failed with %s", clientsetErr)
 			continue
 		}
-		if err := json.Unmarshal([]byte(canaryResp), &canary); err != nil {
-			logger.Errorf("❌ Failed to unmarshal json %s", err)
+
+		nodeList, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+		if err != nil {
+		   logger.Errorf("❗ Get K8s client properties failed with %s", err)
 			continue
 		}
 
-		clusters = append(clusters, api.Cluster{
-			Name: name,
-			Properties: []api.Property{
-				{
-					Name:  "CPU",
-					Icon:  "cpu",
-					Type:  "cpu",
-					Value: "72",
-				},
-				{
-					Name:  "Memory",
-					Icon:  "memory",
-					Type:  "mem",
-					Value: "128",
+		for _, node := range nodeList.Items {
+
+		    properties := MergeNode(node, properties)
+		    	
+			canaryResp, err := net.GET(cluster.CanaryChecker)
+			if err != nil {
+				logger.Errorf("❗ Canary Check failed with %s", err)
+				continue
+			}
+			if err := json.Unmarshal([]byte(canaryResp), &canary); err != nil {
+				logger.Errorf("❗ Failed to unmarshal json %s", err)
+				continue
+			}
+
+			clusters = append(clusters, api.Cluster{
```

Comment:
Move this up into the cluster loop, Nodes are per cluster, each node is not a cluster on its own.